src/js_list.hpp

```diff
@@ -276,6 +278,10 @@ void ListClass<T>::snapshot(ContextType ctx, ObjectType this_object, Arguments&
 {
     args.validate_maximum(0);
     auto list = get_internal<T, ListClass<T>>(ctx, this_object);
+    auto type = list->get_type();
+    if ((type & ~realm::PropertyType::Collection) != realm::PropertyType::Object) {
```

Comment:
```suggestion
    if ((type & ~realm::PropertyType::Flags) != realm::PropertyType::Object) {
```

Should we ignore all flags or just the collections? Does this work for all Collections or just Set and List? I have no idea how Dictionary handles this case...

Reply:
I assume if it doesn't work for collection, it will not work for list and set. I think your suggestion is safer.

Reply:
Note that Collection is just `Array | Set | Dictionary` and `Flags` is (currently) just `Collection | Nullable`. https://github.com/realm/realm-core/blob/65bb2e8d013ef82197f6ea6cc135c7c2a4caa13a/src/realm/object-store/property.hpp#L65-L66